### PR TITLE
add common configuration file for testing data

### PR DESCRIPTION
**Pull request by h-mayorquin**
## Proposal
The idea is to have a common configuration file for the gin test indicating parameters such as :
* The location of the data.
* Whether the outputs are saved or not.
* etc.

# Motivation
While testing in my local machine I usually want a quick way to change the defaults from my local machine. A simple central file `config.py` in the `tests/test_on_data` directory should do the deed.

## To-do
- [x] Modify the documentation if the changes are approved.

## Checklist

- [x] Have you checked our [Contributing](https://github.com/catalystneuro/nwb-conversion-tools/blob/master/docs/contribute.rst) document?
- [x] Have you ensured the PR description clearly describes the problem and solutions?
- [x] Have you checked to ensure that there aren't other open or previously closed [Pull Requests](https://github.com/catalystneuro/nwb-conversion-tools/pulls) for the same change?
- [x] If this PR fixes an issue, is the first line of the PR description `fix #XXX` where `XXX` is the issue number?

### CODE REVIEW

**Comment by CodyCBakerPhD** [comment]
@h-mayorquin Good idea - I would, however, make the name less general than that to avoid confusion with other less specific configuration files (codecov, black, git, etc.).

Maybe just `gin_test_config` would be fine. Also it probably shouldn't be a `.py` file but rather some text storage format like `.json` (which would work well too then because you just extract those two fields from calling `json.load(gin_test_config.json)`.

File: tests/test_on_data/test_YAML_conversion_specification.py [hunk 1]
```diff
@@ -12,20 +12,23 @@
 from nwb_conversion_tools.utils.json_schema import load_dict_from_file
 from nwb_conversion_tools.utils.conversion_tools import run_conversion_from_yaml
 
+# Load data test configuration
+test_config_dic = load_dict_from_file("./tests/test_on_data/gin_test_config.json")
```
Comment (bendichter) [inline tests/test_on_data/test_YAML_conversion_specification.py:16]:
```suggestion
test_config_dict = load_dict_from_file("./tests/test_on_data/gin_test_config.json")
```

File: tests/test_on_data/test_YAML_conversion_specification.py [hunk 2]
```diff
@@ -12,20 +12,23 @@
...
+
 # GIN dataset: https://gin.g-node.org/NeuralEnsemble/ephy_testing_data
 if os.getenv("CI"):
     LOCAL_PATH = Path(".")  # Must be set to "." for CI
     print("Running GIN tests on Github CI!")
 else:
     # Override the LOCAL_PATH to a point on your local system that contains the dataset folder
     # Use DANDIHub at hub.dandiarchive.org for open, free use of data found in the /shared/catalystneuro/ directory
-    LOCAL_PATH = Path("/shared/catalystneuro/")
+    LOCAL_PATH = Path(test_config_dic["LOCAL_PATH"])
```
Comment (bendichter) [inline tests/test_on_data/test_YAML_conversion_specification.py:25]:
```suggestion
    LOCAL_PATH = Path(test_config_dict["LOCAL_PATH"])
```

File: tests/test_on_data/test_gin_ecephys.py [hunk 3]
```diff
@@ -40,20 +41,23 @@
...
-    LOCAL_PATH = Path("/shared/catalystneuro/")
+    LOCAL_PATH = Path(test_config_dic["LOCAL_PATH"])
     print("Running GIN tests locally!")
 
 DATA_PATH = LOCAL_PATH / "ephy_testing_data"
 HAVE_DATA = DATA_PATH.exists()
 
-SAVE_OUTPUTS = False
+SAVE_OUTPUTS = test_config_dic["SAVE_OUTPUTS"]
 if SAVE_OUTPUTS:
```
Comment (bendichter) [inline tests/test_on_data/test_gin_ecephys.py:61]:
```suggestion
if test_config_dic["SAVE_OUTPUTS"]:
```

File: tests/test_on_data/test_gin_ophys.py [hunk 4]
```diff
@@ -18,26 +18,31 @@
...
 
 try:
     from parameterized import parameterized, param
 
     HAVE_PARAMETERIZED = True
 except ImportError:
     HAVE_PARAMETERIZED = False
+
+# Load data test configuration
+test_config_dic = load_dict_from_file("./tests/test_on_data/gin_test_config.json")
```
Comment (bendichter) [inline tests/test_on_data/test_gin_ophys.py:31]:
```suggestion
test_config_dict = load_dict_from_file("./tests/test_on_data/gin_test_config.json")
```

File: tests/test_on_data/test_gin_ophys.py [hunk 5]
```diff
@@ -18,26 +18,31 @@
...
     # Use DANDIHub at hub.dandiarchive.org for open, free use of data found in the /shared/catalystneuro/ directory
-    LOCAL_PATH = Path("/shared/catalystneuro/")
+    LOCAL_PATH = Path(test_config_dic["LOCAL_PATH"])
     print("Running GIN tests locally!")
 OPHYS_DATA_PATH = LOCAL_PATH / "ophys_testing_data"
 HAVE_OPHYS_DATA = OPHYS_DATA_PATH.exists()
 
-SAVE_OUTPUTS = False
+SAVE_OUTPUTS = test_config_dic["SAVE_OUTPUTS"]
 if SAVE_OUTPUTS:
```
Comment (bendichter) [inline tests/test_on_data/test_gin_ophys.py:46]:
```suggestion
if test_config_dic["SAVE_OUTPUTS"]:
```

File: README.md [hunk 6]
```diff
@@ -31,8 +31,8 @@ $ git clone https://github.com/catalystneuro/nwb-conversion-tools
 $ cd nwb-conversion-tools
 $ conda env create -f make_env.yml
 $ conda activate nwb_conversion_env
-$ pip install .
```
Comment (h-mayorquin) [inline README.md:34]: I removed this line because it is unnecessary

Comment (CodyCBakerPhD) [reply README.md:34]: In a separate PR, can you split this section into two options? Option 1, conda install via environment and activation, Option 2, install in current environment via `pip` (and the current PyPI release).

**Comment by CodyCBakerPhD** [comment]
@h-mayorquin Thanks for looking into this! One thing before I can approve:

The lines of `test_config_dict = load_dict_from_file("./tests/test_on_data/gin_test_config.json")` depend on being called from a particular working directory location. This can be relaxed by using a local path relative to each testing file, e.g., something like

```python
test_config_dict = load_dict_from_file(Path(__file__).parent / "gin_test_config.json")
```

**Comment by CodyCBakerPhD** [comment]
@h-mayorquin Also, can you add this new `.json` file to the `.gitignore`, thereby allowing Git to finally be able to ignore anyone's local changes to such things?

**Review by CodyCBakerPhD — APPROVED** [review]

**Pull request merged**